lib/Yield/Calls/Slices.cpp

```diff
@@ -0,0 +1,127 @@
+/// \file Slices.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <unordered_map>
+
+#include "llvm/ADT/BreadthFirstIterator.h"
+#include "llvm/ADT/PostOrderIterator.h"
+
+#include "revng/Yield/Calls/Slices.h"
+
+using NodeView = const yield::Graph::Node *;
+
+struct NodeLookupHelper {
+  const MetaAddress &TargetAddress;
+  bool operator()(NodeView Node) { return Node->Address == TargetAddress; }
+};
```

Comment:
This is only used in one place. Make it a lambda and move it in `preservingSliceImpl` please.

Reply:
Done: https://github.com/revng/revng/pull/191/commits/3e2e0e14a3f81882744de917c956533b738d018c